addons/project/static/tests/project_subtask_kanban_list_tests.js

```diff
@@ -99,17 +99,52 @@ QUnit.module('Subtask Kanban List tests', {
+        await click(target.querySelector('.subtask_list_button'));
+        let subtaskEl = target.querySelector(".subtask_list");
         assert.containsOnce(subtaskEl, ".o_field_widget.o_field_project_task_state_selection.subtask_state_widget_col .o_status:not(.o_status_green,.o_status_bubble)", "The state of the subtask should be in progress");
         await click(subtaskEl, ".o_field_widget.o_field_project_task_state_selection.subtask_state_widget_col .o_status:not(.o_status_green,.o_status_bubble)");
+        subtaskEl = target.querySelector(".subtask_list");
         assert.containsNone(subtaskEl, ".o_field_widget.o_field_project_task_state_selection.subtask_state_widget_col .o_status:not(.o_status_green,.o_status_bubble)", "The state of the subtask should no longer be in progress");
-        assert.containsOnce(subtaskEl, ".o_field_widget.o_field_project_task_state_selection.subtask_state_widget_col .fa-check-circle.text-success", "The state of the subtask should be Done");
         assert.verifySteps([
-            "project.task/search_read",
+            "project.task/web_read",
+            "project.task/onchange",
             "project.task/web_save",
-            "project.task/search_read",
-            "project.task/web_read", // read the parent task to recompute the subtask count
+        ]);
+    });
+
+    QUnit.test("Create a subtask from the kanban view", async function (assert) {
+        let checkSteps = false;
+        const pyEnv = await startServer();
+        const { openView } = await start({
+            serverData: { views: this.views },
+            mockRPC(route, { args, model, method }) {
+                if (checkSteps) {
+                    assert.step(`${model}/${method}`);
+                }
+                if (route === '/web/dataset/call_kw/project.task/create') {
+                    const [{ name, parent_id }] = args[0];
+                    assert.strictEqual(name, 'foo')
+                    assert.strictEqual(parent_id, 1)
+                    return pyEnv['project.task'].create({ name: name, parent_id: parent_id, state: '01_in_progress' });
+                }
+            },
+        });
+        await openView({
+            res_model: "project.task",
+            views: [[false, "kanban"]],
+        });
+
+        assert.strictEqual(target.querySelector(".subtask_list_button").parentElement.textContent, "1/4");
+        checkSteps = true;
+        await click(target.querySelector('.subtask_list_button'));
+        await click(target.querySelector('.subtask_create'));
+        await editInput(target.querySelector('.subtask_create_input'), "input", "foo");
+
+        assert.containsN(target, '.subtask_list_row', 4, "The subtasks list should now display the subtask created on the card, thus we are looking for 4 in total");
+        assert.verifySteps([
+            "project.task/web_read",
+            "project.task/create",
+            "project.task/web_read",
         ]);
```

Comment:
True that for the subtask counter test, thought it was done just above (that's the name of the test) but it really isn't.
Tho i'm pretty sure i would be better of testing this inside a tour, because in qunit tests as i don't have access to the backend i would have to recreate the calculation of the 'closed_subtask_count' of the task1 in the mockRPC which doesn't seem right,
Do you concur or do you think i could do it in the JS test anyway?

yeah btw now the task disappears instantly, updating the state of a task rerender the whole record instantly, i'll try and find a workaround

Reply:
![image](https://github.com/odoo/odoo/assets/36259462/f32fc5c8-c58e-4bf2-9bf0-0e0e8fbf5cc5)
also this

Reply:
Okay so for the first point, regarding the test, you can test it in a tour I don't see a problem with it. 
Now for the second point regarding not being able to change something on the task. I believe I had the same problem when doing the kanban list, but going over the children using the record tag in the xml worked for me. I am not certain whether things should change given the new implementation of the relational model.